doc/object.md

```diff
@@ -1,5 +1,197 @@
 # Object
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+Object is a Javascript object value. A common usecase is to assign many values to a single object. 
+
+Object is extended by [Value](value.md) and extends [Array](array.md), [ArrayBuffer](array_buffer.md), [Buffer<T>](buffer.md), [Function](function.md), and [TypedArray](typed_array.md). 
+
+## Methods
+
+### Empty Constructor 
+
+```cpp
+Napi::Object::Object();
+```
+Creates a new empty Object instance. 
+
+### Constructor
+
+```cpp
+Napi::Object::Object(napi_env env, napi_value value);
+```
+- `[in] env`: The `napi_env` environment in which to construct the Value object.
+
+- `[in] value`: The C++ primitive from which to instantiate the Value. `value` may be any of:
+  - bool
+  - Any integer type
+  - Any floating point type
+  - const char* (encoded using UTF-8, null-terminated)
+  - const char16_t* (encoded using UTF-16-LE, null-terminated)
+  - std::string (encoded using UTF-8)
+  - std::u16string
+  - napi::Value
+  - napi_value
+
+Creates a non-empty Object instance.
+
+### New()
+
+```cpp
+Object Napi::Object::New(napi_env env);
+```
+- `[in] env`: The `napi_env` environment in which to construct the Value object.
+
+Creates a new Object value.
+
+### Set() 
+
+```cpp
+void Napi::Object::Set (____ key, ____ value);
+```
+- `[in] key`: The property that is being assigned a value.
+- `[in] value`: The property that is being assigned to a key.
+
+Assigns the value to the key . 
+
+The key-value pair types can be one of the following:
+- `napi_value` key, `napi_value` value
+- [Value](value.md) key, [Value](value.md) value
+
+Alternatively, the key can be any of the following types:
+- `const char*`
+- `const std::string&`
+- `uint32_t`
+
+While the value must be any of the following types:
+- `napi_value`
+- [Value](value.md)
+- `const char*`
+- `std::string&`
+- `bool`
+- `double`
+
+### Get()
+
+```cpp
+Value Napi::Object::Get(____ key);
+```
+- `[in] key`: The property whose assigned value is being returned.
```

Comment:
How about
returns the value for the property with name `key` on the object.  Returns ?? if the property does not exist. 

We need to see what exactly it returns in that case where the property does not exist and update ??